src/store_client.cc

```diff
@@ -461,7 +486,7 @@ store_client::scheduleMemRead()
     /* What the client wants is in memory */
     /* Old style */
     debugs(90, 3, "store_client::doCopy: Copying normal from memory");
-    size_t sz = entry->mem_obj->data_hdr.copy(copyInto);
+    const auto sz = entry->mem_obj->data_hdr.copy(copyInto); // may be <= 0 per copy() API
     callback(sz);
```

Comment:
AFAICT, data_hdr.copy() always returns a positive answer, but its API is designed to return zero or negative answers as well _and_ its code is quite complex. Eventually, we should adjust that API so that it returns `size_t` instead of `ssize_t` but that adjustment is out of this PR scope. We could also assert that the returned value is positive and convert callback() into noteCopiedBytes(), but I think that would just create a dangerous hidden/unnatural dependency on the data_hdr.copy() method never returning zero.

Instead, this _minimal_ change just removes the silent (and, I bet, accidental) dangerous signed-to-unsigned conversion and explains what is going on.